tools/accuracy_checker/accuracy_checker/main.py

```diff
@@ -266,5 +277,32 @@ def print_processing_info(model, launcher, device, tags, dataset):
     print_info('OpenCV version: {}'.format(cv2.__version__))
 
 
+def write_csv_result(csv_file, processing_info, metric_results):
+    new_file = not check_file_existence(csv_file)
+    field_names = ['model', 'launcher', 'device', 'dataset', 'tags', 'metric_name', 'metric_type', 'metric_value']
+    model, launcher, device, tags, dataset = processing_info
+    main_info = {
+        'model': model,
+        'launcher': launcher,
+        'device': device.upper(),
+        'tags': ' '.join(tags) if tags else '',
+        'dataset': dataset
+    }
+    rows = []
+    for metric_result in metric_results:
+        result = deepcopy(main_info)
+        result.update({
+            'metric_name': metric_result['name'],
+            'metric_type': metric_result['type'],
+            'metric_value': metric_result['value']
+        })
+        rows.append(result)
+    with open(csv_file, 'a+', newline='') as f:
+        writer = DictWriter(f, fieldnames=field_names)
+        if new_file:
+            writer.writeheader()
+        writer.writerows(rows)
```

Comment:
Is it really necessary to accumulate the rows in memory first? You could just move the loop here, and write them into the file one by one.

Reply:
fixed